mantidproject/mantid
# Stop regenerateTracksForEachLambda being hardcoded

**Description of work.**

Correct mistake from commit d48169a8349060223910e0ff7524ba95c17bfdaa where the regenerateTracksForEachLambda parameter of MCAbsorptionStrategy was left hardcoded to False following the creation of a user visible parameter on the MonteCarloAbsorption algorithm which allowed the user to choose.

Setting the user visible value to True was having one minor impact on the calculation. If a value was also supplied for NumberOfWavelengthPoints, it was causing a sparse wavelength grid to be used during the simulation.

Have adjusted some unit tests for tests that were using a value of "true" while actually running the calculation with false. This was either a change to the values being checked or in some cases I've simply switched the test to use the false value because this is the recommended value

Also made minor change to documentation of the NumberOfWavelengthPoints property because I noticed the default described wasn't correct

**To test:**

Run a MonteCarloAbsorption calculation with the ResimulateTracksForDifferentWavelengths parameter set to true and false and observe the results are different. You can use the following script to do this. The number of events per point is set to an unrealistic value of 1 here to keep the runtime down:

```
from __future__ import print_function, division, unicode_literals

from mantid.simpleapi import (ConvertUnits, Load, LoadSampleShape, MonteCarloAbsorption, SetSample, GroupDetectors)
import matplotlib.pyplot as plt

sample_env_mesh = Load(Filename='PEARL00073987.raw', SpectrumList='100')
# MonteCarloAbsorption works in wavelength
sample_env_mesh = ConvertUnits(InputWorkspace=sample_env_mesh, Target='Wavelength')

SetSample(InputWorkspace=sample_env_mesh, Environment={'Name':'Pearl'},Material={'ChemicalFormula': 'Ni'})

sample_env_mesh_abscorr_false = MonteCarloAbsorption(InputWorkspace=sample_env_mesh, EventsPerPoint=1, ResimulateTracksForDifferentWavelengths=False)
sample_env_mesh_abscorr_true = MonteCarloAbsorption(InputWorkspace=sample_env_mesh, EventsPerPoint=1, ResimulateTracksForDifferentWavelengths=True)

CompareWorkspaces(sample_env_mesh_abscorr_false,sample_env_mesh_abscorr_true)
```

Fixes #28504 .

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
